cmd/promtool/tsdb.go

```diff
@@ -446,7 +446,7 @@ func analyzeBlock(path, blockID string, limit int, runExtended bool) error {
 	postingInfos := []postingInfo{}
 
 	printInfo := func(postingInfos []postingInfo) {
-		slices.SortFunc(postingInfos, func(a, b postingInfo) bool { return a.metric > b.metric })
+		slices.SortFunc(postingInfos, func(a, b postingInfo) int { return int(b.metric - a.metric) })
```

Comment:
`b.metric` is unsigned, so you're relying on overflow. Either cite why this works, or cast to `int` before subtracting.